src/webgpu/shader/execution/expression/call/builtin/textureStore.spec.ts

```diff
@@ -120,3 +120,377 @@ Parameters:
+  const size: GPUExtent3D = { width: 1, height: 1, depthOrArrayLayers: 1 };
+  switch (dim) {
+    case '1d':
+      size.width = numTexels;
+      break;
+    case '2d': {
+      const texelsPerArray = numTexels / array;
+      size.width = kWidth;
+      size.height = texelsPerArray / kWidth;
+      size.depthOrArrayLayers = array;
+      break;
+    }
+    case '3d':
+      size.width = kWidth;
+      size.height = numTexels / (2 * kWidth);
+      size.depthOrArrayLayers = 2;
+      break;
+  }
+  return size;
+}
+
+// WGSL declaration type for the texture.
+function textureType(dim: GPUTextureDimension): string {
+  return `texture_storage_${dim}<r32uint, write>`;
+}
+
+// Defines a function to convert linear global id into a texture coordinate.
+function indexToCoord(dim: GPUTextureDimension, type: string): string {
+  switch (dim) {
+    case '1d':
+      return `
+fn indexToCoord(id : u32) -> ${type} {
+  return ${type}(id);
+}`;
+      break;
+    case '2d':
+      return `
+fn indexToCoord(id : u32) -> vec2<${type}> {
+  return vec2<${type}>(${type}(id % width), ${type}(id / width));
+}`;
+      break;
+    case '3d':
+      return `
+fn indexToCoord(id : u32) -> vec3<${type}> {
+  const half = numTexels / depth;
+  let half_id = id % half;
+  return vec3<${type}>(${type}(half_id % width), ${type}(half_id / width), ${type}(id / half));
+}`;
+      break;
+  }
+  return ``;
+}
+
+// Mutates 'coords' to produce an out-of-bounds value.
+// 1D workgroups are launched so 'gid.x' is the linear id.
+function outOfBoundsValue(dim: GPUTextureDimension, type: string): string {
+  switch (dim) {
+    case '1d': {
+      if (type === 'i32') {
+        return `if gid.x % 3 == 0 {
+          coords = -coords;
+        } else {
+          coords = coords + numTexels;
+        }`;
+      } else {
+        return `coords = coords + numTexels;`;
+      }
+      break;
+    }
+    case '2d': {
+      if (type === 'i32') {
+        return `if gid.x % 3 == 0 {
+          coords.x = -coords.x;
+        } else {
+          coords.y = coords.y + height;
+        }`;
+      } else {
+        return `if gid.x % 3 == 1 {
+          coords.x = coords.x + width;
+        } else {
+          coords.y = coords.y + height;
+        }`;
+      }
+      break;
+    }
+    case '3d': {
+      if (type === 'i32') {
+        return `if gid.x % 3 == 0 {
+          coords.x = -coords.x;
+        } else if gid.x % 5 == 0 {
+          coords.y = coords.y + height;
+        } else {
+          coords.z = coords.z + depth;
+        }`;
+      } else {
+        return `if gid.x % 3 == 1 {
+          coords.x = coords.x + width;
+        } else if gid.x % 5 == 1 {
+          coords.y = coords.y + height;
+        } else {
+          coords.z = 2 * depth;
+        }`;
+      }
+      break;
+    }
+  }
+  return ``;
+}
+
+function getMipTexels(numTexels: number, dim: GPUTextureDimension, mip: number): number {
```

Comment:
Is this code correct for the 1d case?

nit: this should get a method comment

 I tried writing a method comment, then got confused.
In the case of `1d`, why don't we have   `texels /= 1<<mip` ?

Because I was going to suggeset

> // Given a number of texels in the base layer of a texture, returns the number of texels in each dimension of the given mip level.

But this code doesn't do that in the `1d` case.